platform/security/sts/security-sts-propertyclaimshandler/src/main/java/org/codice/ddf/security/sts/claims/property/PropertyFileClaimsHandler.java

```diff
@@ -160,7 +169,30 @@ public void setPropertyFileLocation(String propertyFileLocation) {
     if (propertyFileLocation != null
         && !propertyFileLocation.isEmpty()
         && !propertyFileLocation.equals(this.propertyFileLocation)) {
-      userMapping = PropertiesLoader.toMap(PropertiesLoader.loadProperties(propertyFileLocation));
+
+      Properties p = new Properties();
+      try {
+        p.load(getClass().getResourceAsStream(propertyFileLocation));
```

Comment:
I think we should still use the `PropertiesLoader` to load the properties without invoking `toMap` unless the issue is realated to that. `PropertiesLoader.loadProperties(propertyFileLocation)` should return a `Properties`. The advantage of using the `PropertiesLoader` is that it tries to load the properties in multiple ways. See [PropertiesLoader.java#L71](https://github.com/codice/ddf/blob/a888fb0e85bb2c488d87eb16a67e345bf7ea9325/platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java#L71)

Reply:
The PropertiesLoader appeared to be the original cause of the issue. It interprets the users.properties file as a set of strings, but we need it to have a bit more business knowledge and expand the group (\_g\_\:) entries to their individual roles. As we're using the jaas format for users.properties files I thought using the jaas classes would be most appropriate and will keep everything aligned should the jaas format ever be changed.

Reply:
We recommend using the `PropertiesLoader` because the path you get by doing `getClass().getResourceAsStream(propertyFileLocation)` might not always be the correct one. However, I see how in this case the `PropertiesLoader` will return a `java.util.Properties` and to use the `PropertiesBackingEngine` you are looking for an `org.apache.felix.utils.properties.Properties`. 

After talking with @brjeter, to get around this, you can do something similar to the [DefaultUsersDeletionScheduler](https://github.com/codice/ddf/blob/b054ac205251e31390e5ccc9830dfe310a3f2659/platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/DefaultUsersDeletionScheduler.java#L45-L46). Since the location of the `users.properties` file is the same, you can have a static path and only use the setter for tests. You will still need the setter for tests because the `AbsolutePathResolver` won't work as intended for unit tests. This will also mean you won't set the property in the [blueprint](https://github.com/codice/ddf/blob/b054ac205251e31390e5ccc9830dfe310a3f2659/platform/security/sts/security-sts-propertyclaimshandler/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L35) file. Let me know if this isn't clear.

Reply:
@blen-desta I've updated it to include a static path which required a few other changes to check it has been loaded, including updating the unit tests. Let me know if that's what you meant. Did you mean for me to strip it from the blueprint as well?

Reply:
Yes, that looks good. You should delete the line that sets the `propertyFileLocation` in blueprint. https://github.com/codice/ddf/blob/b054ac205251e31390e5ccc9830dfe310a3f2659/platform/security/sts/security-sts-propertyclaimshandler/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L35

Reply:
Thanks. Blueprint file updates added to PR